joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -166,14 +199,16 @@ bool Trajectory::sample(
   return true;
 }
 
-void Trajectory::interpolate_between_points(
+bool Trajectory::interpolate_between_points(
   const rclcpp::Time & time_a, const trajectory_msgs::msg::JointTrajectoryPoint & state_a,
   const rclcpp::Time & time_b, const trajectory_msgs::msg::JointTrajectoryPoint & state_b,
-  const rclcpp::Time & sample_time, trajectory_msgs::msg::JointTrajectoryPoint & output)
+  const rclcpp::Time & sample_time, const bool do_ruckig_smoothing,
+  trajectory_msgs::msg::JointTrajectoryPoint & output)
 {
   rclcpp::Duration duration_so_far = sample_time - time_a;
   rclcpp::Duration duration_btwn_points = time_b - time_a;
 
+  // TODO(anyone): this shouldn't be resized at runtime
```

Comment:
Same deal, per  https://stackoverflow.com/questions/3835857/in-c-does-using-static-variables-in-a-function-make-it-faster/3835902#3835902

```suggestion
```